src/kernel/src/fs/ioctl.rs

```diff
@@ -43,15 +45,17 @@ macro_rules! commands {
                 pub const IOC_OUT: u32 = 0x40000000;
                 pub const IOC_IN: u32 = 0x80000000;
 
-                pub fn try_from_raw_parts(cmd: u64, arg: *mut u8) -> Result<Self, SysErr> {
+                /// # Safety
+                /// `arg` has to be a pointer to the correct value
+                pub unsafe fn try_from_raw_parts(cmd: u64, arg: SysArg) -> Result<Self, SysErr> {
```

Comment:
This is even dangerous than the pointer because it hide the fact that the returned value have a lifetime bind to some pointer.

I guess we should required every PRs that use unsafe to be reviewed by me (or someone else who can catch this kind of bug) so we don't have a dangerous bug get merged. Here is an example of what going to happen if we violate the Rust assumption due to unsafe: https://github.com/obhq/obliteration/issues/48

Reply:
I tried something, take a look

Reply:
Better to use something like this: https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_ptr

Reply:
It's that simple? :D

Reply:
Wait a minute. To be completely honest, sometimes I don't fully understand lifetimes. The CStr struct isn't lifetimed, but our IoCmd and RegMgrCommand are.  Doesn't that make any difference?

Reply:
The point is `try_from_raw_parts` need to have an explicit lifetime and document it how it bind to the pointer.